File: react/my-app/src/services/GradeService.js
import axios from 'axios';

const TEACHER_API_BASE_URL = "http://localhost:9192/api";

class GradeService {

    getGrades(){
        return axios.get(TEACHER_API_BASE_URL + "/grades");
    }

    createGrade(grade){
        return axios.post(TEACHER_API_BASE_URL + "/addgrade" , grade);
    }

    getGradeById(gradeId){
        return axios.get(TEACHER_API_BASE_URL + '/gradeById/' + gradeId);
    }

    getGradesByStudentId(studentId){
        return axios.get(TEACHER_API_BASE_URL + '/gradesByStudentId/' + studentId);
    }

    updateGrade(grade){
        return axios.put(TEACHER_API_BASE_URL + '/updateGrade/', grade);
    }

    deleteGrade(gradeId){
        return axios.delete(TEACHER_API_BASE_URL + '/deleteGrade/' + gradeId);
    }
}

export default new GradeService()

File: languageSchoolProject/src/main/java/com/langschool/languageschoolproject/service/GradeService.java
package com.langschool.languageschoolproject.service;

import com.langschool.languageschoolproject.entity.Grade;
import com.langschool.languageschoolproject.repo.GradeRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class GradeService {

    @Autowired
    private GradeRepo repository;

    public Grade saveGrade(Grade grade)
    {
        return repository.save(grade);
    }

    public List<Grade> saveGrades(List<Grade> grades)
    {
        return repository.saveAll(grades);
    }

    public List<Grade> getGrades()
    {
        return repository.findAll();
    }

    public Grade getGradeById(int id)
    {
        return repository.findById(id).orElse(null);
    }

    public List<Grade> getGradesByStudentId(int id) { return repository.findAll().stream().filter(p -> p.getStudentId() == id).collect(Collectors.toList()); }



    public String deleteGrade(int id)
    {
        repository.deleteById(id);
        return "Grade with id "+id+" removed!";
    }

    public Grade updateGrade(Grade grade)
    {
        Grade existingGrade=repository.findById(grade.getId()).orElse(null);
        assert existingGrade != null;
        existingGrade.setValue(grade.getValue());
        existingGrade.setStudentId(grade.getStudentId());
        existingGrade.setTeacherId(grade.getTeacherId());
        existingGrade.setSubject(grade.getSubject());
        return repository.save(existingGrade);
    }
}
